### DISCUSSION TITLE
Change button visuals when specific routing changes

### DISCUSSION THREAD

**Discussion by L00maca** [oddbear/GoXLR-TouchPortal-Plugin#34] [Q&A] [answered]
I saw that there's a way in the readme to change the text for a button to the status of a GoXLR Routing state.
Is there a way to change the icon (or something else) instead?

Something like an event that triggers every ~500ms like the text change, but just is a container where I can put actions inside such as IF statements, visual change actions or something similar? 

![image](https://user-images.githubusercontent.com/4406596/190032932-56c37af3-81ff-4cbd-8c98-276c2fe42735.png)

**Comment by oddbear** [comment]
Hi.

I am out travelling, so no way for me to test this right now. However, it should be possible I believe.

In the latest pre release (make sure that is the one installed) there should be a page that can be used as an example.

**Comment by L00maca** [comment, marked as answer]
First of all, don't worry and enjoy your travels! 

Anyway I am in the latest pre-release version atm (0.17.1) and I did not find the page you mentioned, but while searching for it I ended up checking your other addons and on AudioMonitor I did find the example of how to use the "when plugin state" event. Turns out that worked perfectly for this. Here's how I did it if anyone in the future ever had the same issue and ends up here:

![image](https://user-images.githubusercontent.com/4406596/190276720-b7828249-4df1-4b95-b104-0fdc64e2d918.png)
